src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractRoslynProject.cs

```diff
@@ -24,8 +29,13 @@ internal abstract partial class AbstractRoslynProject : AbstractProject, ICompil
             string language,
             IServiceProvider serviceProvider,
             VisualStudioWorkspaceImpl visualStudioWorkspaceOpt,
-            HostDiagnosticUpdateSource hostDiagnosticUpdateSourceOpt)
-            : base(projectTracker, reportExternalErrorCreatorOpt, projectSystemName, hierarchy, language, serviceProvider, visualStudioWorkspaceOpt, hostDiagnosticUpdateSourceOpt)
+            HostDiagnosticUpdateSource hostDiagnosticUpdateSourceOpt,
+            string projectFilePath = null,
```

Comment:
Looking through this code in AbstractRoslynProject, none of it seems to be hardened to be called on a background thread. Are we should reusing the existing base is a good way to approach this?